### PR TITLE
homer: init at 24.04.1

### PR DESCRIPTION
**Pull request by huetremy**
## Description of changes

Add homer, a simple dashboard application. This package contains the built static site to be served (eg by nginx).
I added a `package-lock.json` as homer uses `pnpm` and it seems that it is not possible to use this yet #231513 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by huetremy** [comment]
Hi there! This is my first contribution to nixpkgs and I would really appreciate any feedback / tip... to improve !

Moreover, I plan on adding a NixOS module to serve the app. Should I do it in this PR or open another one once this one gets merged ?

**Comment by pluiedev** [comment]
Just FYI, pnpm support is being added in #290715 — for now you can just copy and paste the code from say, [Vesktop](https://github.com/NixOS/nixpkgs/blob/63aceac7660e1fd0c3211ff2399f0d0e7aec868f/pkgs/by-name/ve/vesktop/package.nix#L36-L87) I think

**Comment by pluiedev** [comment]
> Moreover, I plan on adding a NixOS module to serve the app. Should I do it in this PR or open another one once this one gets merged ?

You can add it in this PR and it'd be completely fine

**Comment by huetremy** [comment]
> Just FYI, pnpm support is being added in #290715 — for now you can just copy and paste the code from say, [Vesktop](https://github.com/NixOS/nixpkgs/blob/63aceac7660e1fd0c3211ff2399f0d0e7aec868f/pkgs/by-name/ve/vesktop/package.nix#L36-L87) I think

Thank you for the information! I will follow the same way as Vesktop for now then, and update whenever the pnpm support will be added!

> You can add it in this PR and it'd be completely fine

Perfect, thank you for the quick answer!

**Comment by SuperSandro2000** [comment]
pnpm support should be ready now

**Comment by huetremy** [comment]
Downgraded version for now as 24.05.1 requires more recent pnpm (package.json says 9.1.4, error message says 9.1.2 when building v24.05.1).

I will write the nixos module and then upgrade when newer pnpm is available

File: pkgs/by-name/ho/homer/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ stdenvNoCC, lib, buildNpmPackage, fetchFromGitHub, nodejs, pnpm_8 }:
```
Comment (FliegendeWurst) [inline pkgs/by-name/ho/homer/package.nix:1]: Please format with the new formatter.

File: pkgs/by-name/ho/homer/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    runHook preInstall
+
+    mkdir -p $out/lib
+    cp -R ./dist/* $out/lib/
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A full static html/js dashboard, based on a simple yaml configuration file";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ho/homer/package.nix:41]:
```suggestion
    description = "Static html/js dashboard, based on a simple yaml configuration file";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by gepbird** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/354293

**Pull request closed without merging**